hazelcast/src/main/java/com/hazelcast/spi/discovery/multicast/impl/MulticastDiscoverySender.java

```diff
@@ -67,15 +67,22 @@ private void initDatagramPacket() throws IOException {
     public void run() {
         while (!stop) {
             try {
-                Thread.sleep(SLEEP_DURATION);
-            } catch (InterruptedException e) {
-                logger.finest("Thread sleeping interrupted. This may due to graceful shutdown.");
-            }
-            try {
                 send();
             } catch (IOException e) {
                 logger.finest(e.getMessage());
             }
+            sleepUnlessStopped();
+        }
+    }
+
+    private void sleepUnlessStopped() {
+        if (stop) {
+            return;
+        }
+        try {
+            Thread.sleep(SLEEP_DURATION);
```

Comment:
Isn't this `SLEEP_DURATION ` too long? in multicast joiner, we use a random delay between 50ms - 200ms.

Reply:
this is a sender. we probably do not want to flood network with multicast packets every 50ms

Reply:
hmm... `MulticastDiscoverySender` continues broadcasting packets even after cluster is formed. That looks like a very naive implementation.

Reply:
indeed. however fixing this is not my goal for this PR.